src/main/java/seedu/address/logic/commands/ListCommand.java

```diff
@@ -5,30 +5,155 @@
 
 import java.util.HashMap;
 import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Objects;
 import java.util.function.Predicate;
+import java.util.stream.Collectors;
 
 import seedu.address.logic.parser.Prefix;
+import seedu.address.model.FriendlyLink;
 import seedu.address.model.Model;
 import seedu.address.model.pair.Pair;
 import seedu.address.model.person.Elderly;
 import seedu.address.model.person.Volunteer;
 
 /**
- * Lists all elderly, volunteers and pairs in FriendlyLink.
+ * Lists all elderly, volunteers and pairs in FriendlyLink based on the action given.
  */
 public class ListCommand extends Command {
 
     public static final String COMMAND_WORD = "list";
     public static final HashMap<Prefix, String> COMMAND_PROMPTS = new LinkedHashMap<>();
-    public static final String MESSAGE_SUCCESS = "Listed all elderly, volunteers and pairs";
+    public static final String MESSAGE_SUCCESS = "Listed all volunteers, elderly and pairs";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": List all volunteers and elderly pairs if nothing is"
+            + "specified after the command, else list all paired or unpaired volunteers and elderly if the"
+            + "\"paired\" or \"unpair\" (case-insensitive) word is given respectively after the the list command word"
+            + "Parameters: [paired/unpair]\n"
+            + "Example: " + COMMAND_WORD + " paired";
+    public static final String MESSAGE_SUCCESS_LIST_PAIRED =
+            "Listed %1$s paired volunteer, %2$s paired elderly and all pairs";
+    public static final String MESSAGE_SUCCESS_LIST_UNPAIRED =
+            "Listed %1$s unpaired volunteer, %2$s unpaired elderly and all pairs";
```

Comment:
I think a better behaviour would be to not show any pairs at all when listing unpaired volunteers and elderly. What do you think?

Reply:
List all pairs is better, there's no downside to it if they need to refer to the pair list will going through pair or unpaired people. On the other hand, not showing any pairs would by chance cause inconvenience if they need it.

Reply:
Yup, make sense.